cpp/src/gandiva/cache.cc

```diff
@@ -20,13 +20,14 @@
 
 namespace gandiva {
 
-static const int DEFAULT_CACHE_SIZE = 500;
+static const size_t DEFAULT_CACHE_SIZE = 128 * 1024 * 1024;  // 256 MiB
 
-int GetCapacity() {
-  int capacity;
+size_t GetCapacity() {
+  size_t capacity;
   const char* env_cache_size = std::getenv("GANDIVA_CACHE_SIZE");
   if (env_cache_size != nullptr) {
-    capacity = std::atoi(env_cache_size);
+    capacity = std::stoul(env_cache_size);
```

Comment:
why is this change required?

Reply:
In atoi, if no conversion happens, because of an invalid argument or it falls out of range, the value will assume the zero value, in stoul if there is a problem it will throw an excpetion leading to abetter error management.

Reply:
atoi was specifically added because it doesn't throw exception. The zero case is handled in next line.
Also in this change you are not catching the exception so the change is incorrect. Better just undo this change.

Reply:
Ok, I changed it back.